apps/financial-aid/web-veita/src/routes/MySettings/MySettings.tsx

```diff
@@ -1,39 +1,62 @@
-import React, { useContext, useState } from 'react'
+import React, { useContext, useEffect, useState } from 'react'
 
-import {
-  ApplicationOverviewSkeleton,
-  LoadingContainer,
-} from '@island.is/financial-aid-web/veita/src/components'
 import {
   Text,
   Box,
   Input,
   ToggleSwitchCheckbox,
+  Button,
+  toast,
+  ToastContainer,
 } from '@island.is/island-ui/core'
 import { AdminContext } from '@island.is/financial-aid-web/veita/src/components/AdminProvider/AdminProvider'
 
 import * as profileStyles from '@island.is/financial-aid-web/veita/src/components/Profile/Profile.css'
-import { InputType, isEmailValid } from '@island.is/financial-aid/shared/lib'
+import {
+  InputType,
+  isEmailValid,
+  User,
+} from '@island.is/financial-aid/shared/lib'
+import { useLazyQuery, useMutation } from '@apollo/client'
+import {
+  CurrentUserQuery,
+  UpdateStaffMutation,
+} from '@island.is/financial-aid-web/veita/graphql'
 
 interface mySettingsState {
   nationalId?: string
   name?: string
   email?: string
   nickname?: string
+  pseudonymName: boolean
   hasError: boolean
   hasSubmitError: boolean
 }
 
 export const MySettings = () => {
-  const { admin } = useContext(AdminContext)
+  const { admin, setAdmin } = useContext(AdminContext)
+
+  const [updateStaff, { loading }] = useMutation(UpdateStaffMutation)
+
+  const [getCurrentUser, { data }] = useLazyQuery<{
+    currentUser: User
+  }>(CurrentUserQuery, {
+    fetchPolicy: 'no-cache',
+    errorPolicy: 'all',
+  })
```

Comment:
On this screen we should not have the option to update nationalId and name, so let's start taking that out, after that is done then you can return staff from updateStaff